### ISSUE TITLE
Should PM rooms only display in the PMList if there are messages within that room?

### ISSUE THREAD

**Issue by lisjak** [zoot-hq/zoot#206]
- With the current set-up, if I (user1) create a new PM with a user (user2), it will display in user2's PM list even if I haven't sent a message. user2 will see an empty PM room (or the entry/exit stamps, but those need to be removed.)
- Should we change it so that PM rooms only appear in the PMList of either user if the PM room contains messages?
- Otherwise, users will have a list of empty rooms in which no messages are contained. They will see these rooms even if they weren't the ones who initiated the PM.
- This could make for awkward encounters if you accidentally press a username in chat, especially since we do not have functionality to hide or delete PM rooms.

**Comment by lkobetz** [comment]
Since we don't have the functionality to hide or delete rooms, I think you're right. But if that functionality were available, I think we should show all rooms since the user probably would have had a reason to start a PM with that person even if they didn't write anything.
Additionally, maybe when a user clicks on another user's name, a popup could appear asking them if they'd like to start a new private chat with that user. Especially since, when people first start using the app, they might just try pressing everything to see what it does, or they might expect something different to happen when they press the user's name.

**Comment by lisjak** [comment]
> Since we don't have the functionality to hide or delete rooms, I think you're right. But if that functionality were available, I think we should show all rooms since the user probably would have had a reason to start a PM with that person even if they didn't write anything.
> Additionally, maybe when a user clicks on another user's name, a popup could appear asking them if they'd like to start a new private chat with that user. Especially since, when people first start using the app, they might just try pressing everything to see what it does, or they might expect something different to happen when they press the user's name.

- I think the popup is a great idea, and will reduce the need for hiding chatroom. 
- I don't think we should have a way to delete PM rooms because I feel like that gives users too much power, so to speak. A situation could arises where one user could delete a PM even if the other user didn't want that to happen. 
- Giving users the option to hide PM rooms within their PM list is a good idea. 
- The problem with empty PMs is that they appear on both user's ends. If user1 starts a PM with user2 and never sends anything, the user1-user2 room will appear in user2's chat list. I think this would be confusing for user2.  Even with hiding functionality, it might be best not to display empty rooms, but I'm open to more thoughts on it.

**Comment by lkobetz** [comment]
Yeah, I think in this case maybe you're right that a PM room should only appear if it has messages. Maybe there could also be a way to allow the user to 'favorite' or 'bookmark' PMrooms they're particularly interested in, or sort the list of rooms by date of most recent message to keep the most recently used rooms at the top, effectively 'hiding' rooms without much action.
